posthog/src/main/java/com/posthog/PostHogOkHttpInterceptor.kt

```diff
@@ -0,0 +1,52 @@
+package com.posthog
+
+import com.posthog.internal.RRPluginEvent
+import com.posthog.internal.capture
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+
+public class PostHogOkHttpInterceptor(private val captureNetwork: Boolean = true) : Interceptor {
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val originalRequest = chain.request()
+
+        try {
+            val response = chain.proceed(originalRequest)
+
+            captureNetworkEvent(originalRequest, response)
+
+            return response
+        } catch (e: Throwable) {
+            throw e
+        }
+    }
+
+    private fun captureNetworkEvent(request: Request, response: Response) {
+        if (!captureNetwork) {
+            return
+        }
+        val url = request.url.toString()
+        val method = request.method
+        val statusCode = response.code
+        val start = response.sentRequestAtMillis
+        val end = response.receivedResponseAtMillis
+        var transferSize = (response.body?.contentLength() ?: 0) + (request.body?.contentLength() ?: 0)
+
+        if (transferSize < 0) {
+            transferSize = 0
+        }
+
+        val requestMap = mapOf<String, Any>(
+            "name" to url,
+            "method" to method,
+            "status" to statusCode,
+            "timestamp" to end,
+            "duration" to (end - start),
+            "transferSize" to transferSize,
+        )
```

Comment:
(will check)

Reply:
I think it needs to be `responseStatus`

Reply:
Yeah, I believe this could be inferred from either a content type, mime type, or URL, but for now let's stick to `fetch`, which is not wrong, but very generic =D
I check the `response.cacheResponse` and set it to 0 if coming from response.

> If you will always be initiator type fetch - or almost always - then I'll default it in the UI/ingestion to save wasted bytes

Sounds good, since we don't have a "fetch/XHR" API, but everything is a request, so technically everything can be a `fetch` as well.
Maybe the filtering can be more clever when headers are there, so you can inspect content type.